### ISSUE TITLE
Tree Select Dropdown doesn't allow outside clicks

### ISSUE THREAD

**Issue by stumitchell** [day8/re-com#343]
Tree select Dropdown doesn't allow the user to click on html elements outside the component while the drop-down is showing.

This is in contrast to single drop-down which removes the dropdown and accepts the click on, for example, a button.

Unfortunately this means the user currently has to click outside once to close the dropdown and then again to click on the button.

**Comment by Gregg8** [comment]
FYI, this is because `tree-select` uses `popover` which uses an internal backdrop component that covers the entire screen that uses `on-click` to close the popover.

`single-dropdown` however, uses `on-blur` to close the dropdown part.

Therefore you'd need to modify `popover` to using `on-blur` for closing it. I would recommend having an option to specify the close method for backward compatibility.

**Comment by kimo-k** [comment]
Hey @stumitchell, I made some changes which you can try out in `2.21.17`. Clicking outside the anchor/body will now register in the rest of the DOM. 

The `:backdrop` part is now purely visual, and a global event handler now handles the closing behavior.

**Comment by kimo-k** [comment]
> tree-select uses popover which uses an internal backdrop component that covers the entire screen that uses on-click to close the popover.

It did use a backdrop component, but not `popover`. That said, I noticed some weird naming that might have suggested that to you. I've changed those names, now.

> Therefore you'd need to modify popover to using on-blur for closing it.

The problem with using `:on-blur` is that it's hard to control when the blur event will trigger. Interacting with elements inside the anchor or body seems to trigger blur events, making components like `tree-select-dropdown` unusable. It might be possible to get this all working correctly, but so far, it all seems hard to generalize.

I liked the simplicity of using the backdrop component, but I see the problems it caused for clicking & scrolling outside the `dropdown` component. 

> I would recommend having an option to specify the close method for backward compatibility.

This new method seems alright. Of course, more issues could arise. For now, I hope we can avoid adding optional behavior.
